### DISCUSSION TITLE
Error: no tab content was found with activate_tab_id 'SC-tab-main-shell-commands' (conflict with Settings Search plugin)

### DISCUSSION THREAD

**Discussion by luckman212** [Taitava/obsidian-shellcommands#348] [Debugging and testing]
SC 0.18.2

I was debugging something else, and I noticed on every startup 2 errors logged to the console by SC:

<img width="698" alt="image" src="https://user-images.githubusercontent.com/1992842/236693026-fce3d6f0-bc2a-4ee2-a966-4b936c0c8ccc.png">

![image](https://user-images.githubusercontent.com/1992842/236692892-da4866d7-f81b-4069-ba6d-c0cd0cf68b53.png)

This only happens with Shell Commands and only when Settings Search is enabled. It looks like it's caused by the `buildPluginResources() -> getTabResources()` function in Settings Search being unable to iterate over one of the pluginTabs of SC:

```js
  buildPluginResources() {
    const tab = this.app.setting.pluginTabs[this.pluginTabIndex];
    if (tab) {
      this.getTabResources(tab);
      this.pluginTabIndex++;
      setTimeout(() => this.buildPluginResources());
    }
  }
```

```js
 async getTabResources(tab) {
    await tab.display();
    const settings = tab.containerEl.querySelectorAll(".setting-item:not(.setting-item-header)");
    for (const el of Array.from(settings)) {
      const text = el.querySelector(".setting-item-name")?.textContent;
      if (!text)
        continue;
      const desc = el.querySelector(".setting-item-description")?.innerHTML ?? "";
      const resource = {
        tab: tab.id,
        name: tab.name,
        text,
        desc
      };
      this.resources.push(resource);
      this.addResourceToCache(resource);
    }
    if (this.app.setting.activeTab?.id == tab.id)
      return;
    tab.containerEl.detach();
    tab.hide();
  }
```

**Comment by Taitava** [comment]
Hi! Thank you for your report. I appreciate the details you provided and research you've done 🙂 . Unfortunately I don't have much time for this project at the moment, so I can't say when I can look into this.

**Comment by luckman212** [comment]
That's ok. I know the feeling! I have about 10 projects I've been meaning to get around to for the last 6 months. Hard to find time these days.
